### ISSUE TITLE
Issue Formatting

### ISSUE THREAD

**Issue by byklyci** [bounswe/bounswe2018group9#3]
_Due by Feb 12,2018_ 

We are trying to understand the best format of issue. Mastering issues should include as follows:
*A title and description describe what the issue is all about.
*Color-coded labels help you categorize and filter your issues (just like labels in email).
*A milestone acts like a container for issues. This is useful for associating issues with specific features or 
 project phases (e.g. Weekly Sprint 9/5-9/16 or Shipping 1.0).
*One assignee is responsible for working on the issue at any given time.
*Comments allow anyone with access to the repository to provide feedback. 
*Notification, @mentions and references.
*Search
*Overviews&Reports 
*Other uses for issues.

[Mastering Issues · Github Guides](https://guides.github.com/features/issues/)

![under_maintenance_dribbble_1x](https://user-images.githubusercontent.com/23294453/35938631-beb142f2-0c5a-11e8-8d34-5ab862637a64.png)

Please add your comments...

**Comment by nazmicancalik** [comment]
One more point to add:

If there is an issue that is going to be assigned to specific person, mention his/her username in the 
Issue title. Only assignee part is not enough.

**Comment by kemaltulum** [comment]
I agree with Nazmi, which is why I heard from the ones having taken this course that Suzan Hoca likes mentioning the assignee names in the title. Still it is arguable. We can embrace another way if we do not like doing in this way

**Comment by okyksl** [comment]
@nazmicancalik I think if issue title should focus on problem/task more, and that would be clearer way of conveying information. One problem associated with your suggestion is that some issues will be assigned to more than one person, in that case issue title will be a bit messy.

On Issue page, you can either click on assignee and then filter a user to search assignees. Further, you can use advanced searching as discussed on this resource:

> https://help.github.com/articles/searching-issues-and-pull-requests/

**Comment by byklyci** [comment]
The idea Of adding name in issue title will be efficient. But in my opinion; whoever and whatever written in the title, the others should check the issue everytime.

**Comment by nazmicancalik** [comment]
I added the issue format file. For now it doesn't say we have to add assignee's to the title. We can add the name specification according to the group consensus.

**Comment by byklyci** [comment]
If you wont add anything about issue formating, ı will close the issue. @nazmicancalik

**Comment by byklyci** [comment]
I'm closing the issue until the specifics of the project becomes clear. After that, we probably need to discuss additional set of settings to cover cases such as implementation platform.

**Comment by okyksl** [comment]
@byklyci let's organize your summary & guide and @nazmicancalik Issue Template, as a Issue Contribution Guide and load it to the [Wiki](https://github.com/bounswe/bounswe2018group9/wiki/Issues)

**Comment by byklyci** [comment]
I reopened that issue because our issue summary and guidelines need same specific flow.So can you say your opinion about our issue formatting!Then ı will edit it.

**Comment by nazmicancalik** [comment]
I think the last consensus was following: 
* There won't be any names included in issue titles.
* Issue title grammar should be in an imperative format. (__Ex:__ ~~Deleting~~ index.html, Delete index.html )

Any more thoughts ?

**Comment by kemaltulum** [comment]
@nazmicancalik Imperative format would not be so convenient for issues in such cases, for example, when reporting an error, you would probably put the error message to the title. On the other hand, most of the issues we have right now is in a more flexible format like this one "Issue Formatting"
There would be a replacement for this title for me in imperative form "Discuss Issue Formatting (Rules)"
In short, making all the titles imperative could be cumbersome. 
The critical thing is that the title shall be explanatory and identifiable.

**Comment by byklyci** [comment]
@kemaltulum you are right because of title and the number of issue has a huge impact.

**Comment by serkan-ozel-2015400123** [comment]
@nazmicancalik Imperative form will be used in commit messages but not in issue titles, I remember. So if we are not going to set a certain issue form for its content (not its title), we can close this issue.

**Comment by okyksl** [comment]
@serkan-ozel-2015400123 It means that don't open any issue such as: `Serkan: Do This`, instead just use `Do This` with assignee `Serkan`.

**Comment by nazmicancalik** [comment]
@serkan-ozel-2015400123 I think everything is clear after @okyksl 's reply. I think @serkan-ozel-2015400123 is right we can close this issue. However, we should put the consensus ( no names and imperative ) to a related wiki page. What do you think ? @okyksl @serkan-ozel-2015400123 @byklyci

**Comment by nazmicancalik** [comment]
@byklyci Yusuf, issue page on the sidebar seemed a bit long to me what do you say we make it a bit smaller? What do you think ?

**Comment by nazmicancalik** [comment]
@kemaltulum Kemal I just saw your feedback. You have your points, I couldn't decide. We can argue this on next week's meeting to get the consensus. You can add an agenda item about this to the body of issue #26 .

**Comment by byklyci** [comment]
If you are all agree about closing this issue ı will do it.

**Comment by okyksl** [comment]
@nazmicancalik @byklyci @serkan-ozel-2015400123 @kemaltulum Let's use this issue to track the **ideas** about the `Issue Formatting`. As @nazmicancalik said, we should definitely put ideas cultivated here to the `Wiki`, _but the completion of that guide is another task_ - so let's close it.

**Comment by serkan-ozel-2015400123** [comment]
@byklyci You can close it, please 👍
